synergy-design-system/synergy-design-system
# fix: 🤔 Update project dependencies

# Pull Request

## 📖 Description

This PR updates all possible packages to the latest version.

> ⚠️ I have made an exception for `semantic-release-monorepo`, which got new updates in v7 and also a new version v8 entirely, which now also supports esm. I would like to create a ticket for this as it is workflow critical for our builds. This will be addressed in #277.

### 🎫 Issues

Closes #258 

## 👩‍💻 Reviewer Notes

As we have substantial updates to nearly every package, everything has to be checked (e.g. linting, building, tests, framework wrappers, docs).

I also opted to remove some unneeded dependencies:
- assets package, removed style-dictionary in the package.json without requiring it.
- components package, removed vue-esbuild plugin and the vue/esbuild step completely as it is not needed at all. Also lit/react is not needed, so it was also removed.
- docs does not use the badges plugin, so I removed it.

Components errored out some stylelint errors after upgradeing it, so I fixed those issues, too (this is the reason this is a "fix" and not a "chore".

## 📑 Test Plan

- Check out this branch and run `pnpm i -r && pnpm build && pnpm lint`. There should be no errors
- Make sure icons are drawn via `cd packages/assets && pnpm fetch-assets`. There should be no diffs between icons!
- Check the components package via `cd packages/components && pnpm vendor.get && pnpm vendor.set && pnpm build && pnpm test && pnpm lint`. There should be no issues, no new files in components or the wrappers.
- Check if Docs work via `cd packages/docs && pnpm start`. Navigate around and see if there are any issues arising in the started application.
- Check if the framework wrappers are still working by starting the local demos and playing around with them.

## ✅ DoD
